packages/react-app/src/App.jsx

```diff
@@ -13,29 +13,55 @@ import externalContracts from 'contracts/external_contracts'
 import { switchToOptimism } from 'helpers/SwitchToOptimism'
 const { ethers } = require('ethers')
 
+const APPSTATEACTION = {
+  GOERLICHAINID: '5',
+  OPTIMISMCHAINID: '10',
+}
+
+const defaultState = {
+  provider: new ethers.providers.Web3Provider(window.ethereum),
+  chainId: '5',
+  contractRef: externalContracts['5'].contracts.REMIX_REWARD,
+}
+
+function appStateReducer(state, actionType) {
+  if (actionType.GOERLICHAINID) {
```

Comment:
there isn't `actionType.OPTIMISMID` here too?

Reply:
There is, OPTIMISMCHAINID on line 18.